### PR TITLE
CommunicationPlan copy

### PR DESCRIPTION
**Pull request by streeve**
First, adding a test which fails in copying the `CommunicationPlan` due to duplicating communicators. Add missing copy constructor/assignment operator.

Found during periodic communication #263 with @sslattery 

Increased coverage by explicitly setting `MPI_EXEC_NUMPROCS=2` for travis (previously running everything one one rank)

### CODE REVIEW

File: core/unit_test/tstCommunicationPlan.hpp [hunk 1]
```diff
@@ -575,6 +575,12 @@ void test7( const bool use_topology )
         EXPECT_EQ( n, host_steering( n ) );
 }
 
+void test_copy()
+{
+    Cabana::CommunicationPlan<TEST_DEVICE> comm_plan( MPI_COMM_WORLD );
```
Comment (sslattery) [inline core/unit_test/tstCommunicationPlan.hpp:580]: I would say this test is not sufficient - we want to be able to check that the state of the `CommunicationPlan` has not changed either. The operators you have written only duplicate the comm but we will also will need them to copy over the rest of the state (e.g. all of the `std::vector` objects, etc.)

**Review by sslattery — CHANGES_REQUESTED** [review]

File: core/src/Cabana_CommunicationPlan.hpp [hunk 2]
```diff
@@ -300,6 +300,23 @@ class CommunicationPlan
         MPI_Comm_dup( comm, &_comm );
     }
 
+    /*!
+      \brief Copy constructor.
+    */
+    CommunicationPlan( const CommunicationPlan &cp )
```
Comment (dalg24) [inline core/src/Cabana_CommunicationPlan.hpp:306]: You need to copy these guys too
https://github.com/ECP-copa/Cabana/blob/a8554efcfae9168e83868a173de58faecc7f8fc1/core/src/Cabana_CommunicationPlan.hpp#L821-L827

File: core/unit_test/tstCommunicationPlan.hpp (hunk 1 above)
Comment (streeve) [reply core/unit_test/tstCommunicationPlan.hpp]: I copied inside an existing test instead - I can add more or check each member individually

File: core/src/Cabana_CommunicationPlan.hpp (hunk 2 above)
Comment (streeve) [reply core/src/Cabana_CommunicationPlan.hpp:306]: Done

File: core/src/Cabana_CommunicationPlan.hpp [hunk 3]
```diff
@@ -300,6 +300,37 @@ class CommunicationPlan
...
+    {
+        MPI_Comm_dup( cp.comm(), &_comm );
+        _neighbors = cp._neighbors;
+        _total_num_export = cp._total_num_export;
+        _total_num_import = cp._total_num_import;
+        _num_export = cp._num_export;
+        _num_import = cp._num_import;
+        _num_export_element = cp._num_export_element;
+        _export_steering = cp._export_steering;
```
Comment (dalg24) [inline core/src/Cabana_CommunicationPlan.hpp:315]: We need to discuss that copying the view.  If it was a deep copy, then `CommunicationPlan` would have value semantics.  But with that lien it neither has value nor reference semantics which makes it harder to reason about.

**Comment by streeve** [comment]
Updated to match ArborX/ArborX#369

**Comment by streeve** [comment]
It looks like the codecov patch is failing because we're only running with one MPI rank on travis (and therefore not testing a reasonable fraction of the `CommPlan`). @junghans is there an easy way to get this to run on at least 2 MPI ranks?

**Comment by junghans** [comment]
Try to set `MPIEXEC_MAX_NUMPROCS=2`.

File: core/src/Cabana_CommunicationPlan.hpp [hunk 4]
```diff
@@ -460,11 +466,11 @@ class CommunicationPlan
 
         // Get the size of this communicator.
         int comm_size = -1;
-        MPI_Comm_size( _comm, &comm_size );
+        MPI_Comm_size( *_comm_ptr, &comm_size );
```
Comment (dalg24) [inline core/src/Cabana_CommunicationPlan.hpp:469]:
```suggestion
        MPI_Comm_size( comm(), &comm_size );
```

Comment (streeve) [reply core/src/Cabana_CommunicationPlan.hpp]: Fixed

File: core/src/Cabana_CommunicationPlan.hpp [hunk 5]
```diff
@@ -762,7 +768,7 @@ class CommunicationPlan
...
```
Comment (dalg24) [inline core/src/Cabana_CommunicationPlan.hpp:771]:
```suggestion
        MPI_Comm_size( comm(), &comm_size );
```

**Comment by streeve** [comment]
> Try to set `MPIEXEC_MAX_NUMPROCS=2`.

@junghans I added to Travis, but it didn't seem to change anything

**Comment by junghans** [comment]
> > Try to set `MPIEXEC_MAX_NUMPROCS=2`.
> 
> @junghans I added to Travis, but it didn't seem to change anything

fixed

**Comment by streeve** [comment]
@sslattery this is ready for re-review

**Review by sslattery — APPROVED** [review]

**Pull request merged**